services-directions-models/src/test/java/com/mapbox/api/directions/v5/models/DirectionsResponseTest.java

```diff
@@ -41,15 +49,27 @@ public void fromJson_correctlyBuildsFromJson() throws Exception {
   }
 
   @Test
-  public void testToFromJson() throws Exception {
-    String json = loadJsonFixture(DIRECTIONS_V5_PRECISION6_FIXTURE);
-    DirectionsResponse responseFromJson1 = DirectionsResponse.fromJson(json);
+  public void testToFromJsonWithRealResponse() throws Exception {
+    Gson gson = new GsonBuilder().create();
+    String originalJson = loadJsonFixture(DIRECTIONS_V5_PRECISION6_FIXTURE_ARTIFICIAL_FIELDS);
+    JsonObject originalJsonObject = gson.fromJson(originalJson, JsonObject.class);
+
+    DirectionsResponse model = DirectionsResponse.fromJson(originalJson);
+    JsonObject jsonFromObject = gson.fromJson(model.toJson(), JsonObject.class);
+
+    assertEquals(originalJsonObject, jsonFromObject);
```

Comment:
```suggestion
    DirectionsResponse model = DirectionsResponse.fromJson(originalJson);
    String jsonFromModel = model.toJson();

    assertEquals(originalJson, jsonFromModel);
```
We want to ensure the resulting strings are correct, the object wrapping step seems unnecessary.

Reply:
Strings won't the same. An order of fields changes when json gets serialised to java model and then gets deserialised back. And I think it's acceptable behaviour. `JsonObject` compares that 2 JSONs has the same set of fields with the same value and it doesn't care about order.

Reply:
That makes sense :+1: I guess the equality of a serialized form is just a side effect, not a guarantee.